src/api/spec/services/scm_status_reporter_spec.rb

```diff
@@ -37,4 +37,37 @@
       it { expect(subject.state).to eq('failed') }
     end
   end
+
+  describe '#call' do
+    context "repository doesn't exist" do
+      let!(:user) { create(:confirmed_user, :with_home, login: 'jane_doe') }
+      let!(:package) { create(:package, name: 'bye', project: user.home_project) }
+
+      let(:workflow_token) { create(:workflow_token, user: user) }
+      let(:token) { workflow_token.scm_token }
+
+      let(:event_payload) do
+        { project: user.home_project_name, package: package.name, repository: 'openSUSE_Leap', arch: 'x86_64' }
+      end
+
+      let(:event_subscription_payload) { { scm: 'github' } }
+
+      let(:event_type) { 'Event::BuildSuccess' }
+
+      let!(:event_subscription) do
+        EventSubscription.create(channel: 'scm', package: package, eventtype: event_type, receiver_role: 'reader', token: workflow_token)
+      end
+
+      let(:octokit_client) { Octokit::Client.new }
+
+      subject { scm_status_reporter.call }
+
+      before do
+        allow(Octokit::Client).to receive(:new).and_return(octokit_client)
```

Comment:
```suggestion
```
This `allow` sentence is not needed. `octokit_client` is already defined with a `let` in line 61.

Reply:
Without this, we couldn't mock `Octokit::Client` to raise `Octokit::InvalidRepository` when `create_status` is called. This happens on the next line.

Reply:
Funny thing is that the test passes anyways.

Reply:
Exactly @dmarcoux. Eduardo as I wrote you privately, the test passes because the repository is wrong anyway. But that's not deterministic. Forcing it to a specific state is deterministic and avoid any flaky condition.

Reply:
Now I get it. Thanks for the explanations.

Maybe it could help to check for an invalid token, also. I guess if Octokit also raises an `Octokit::InvalidRepository` exception when the token is invalid.